cpinstaller/src/cpinstaller.js

```diff
@@ -391,466 +792,371 @@ export class CPInstallButton extends InstallButton {
+        // command behind an if statement, this will not detect it is enabled.
+        let fileContents;
+        if (this.repl) {
+            return true; // Always disabled in this case
+        } else if (this.circuitpyDriveHandle) {
+            fileContents = await this.readFile("boot.py");
+            // TODO: Compare board's boot.py to our boot.py by
+            // searching for storage.disable_usb_drive() at the beginning of the line
+        } else {
+            this.errorMsg("Connect to the CIRCUITPY drive or the REPL first");
+            return {};
+        }
 
-function hasBoard(board) {
-    for (let opt of binSelector.options) {
-        if (opt.value == board) { return opt }
+        if (fileContents) {
+            return toml.parse(fileContents);
+        }
+        this.logMsg("Unable to read settings.toml from CircuitPython. It may not exist. Continuing...");
+        return {};
     }
-}
 
-function toggleConsole(show) {
-    // hide/show the console log and its widgets
-    const consoleItemsMethod = show ? "remove" : "add"
-    for (let idx = 0; idx < consoleItems.length; idx++) {
-        consoleItems.item(idx).classList[consoleItemsMethod]("hidden")
-    }
-    // toggle the button
-    //butShowConsole.checked = show
-    // tell the app if it's sharing space with the console
-    const appDivMethod = show ? "add" : "remove"
-    appDiv.classList[appDivMethod]("with-console")
-
-    // scroll both to the bottom a moment after adding
-    setTimeout(() => {
-        log.scrollTop = log.scrollHeight
-        appDiv.scrollTop = appDiv.scrollHeight
-    }, 200)
-}
-
-let semver
-function initSemver(newSemver) {
-    if (!newSemver) { return }
-
-    semver = newSemver
-    semverLabel.innerHTML = semver
-
-    return true
-}
-
-function lookupFirmwareByBinSelector() {
-    // get the currently selected board id
-    const selectedId = binSelector.value
-    if (!selectedId || selectedId === 'null') { throw new Error("No board selected.") }
-
-    // grab the stored firmware settings for this id
-    let selectedFirmware
-    for (let firmware of latestFirmwares) {
-        if (firmware.id === selectedId) {
-            selectedFirmware = firmware
-            break
+    saveBootDisabled(disabled) {
+        // TODO: Save/remove a copy of boot.py on the CIRCUITPY drive
+        // This depends on whether it is currently disabled in boot.py and what the value of disabled is
+        // If they are the same, we can skip
+        // An idea is to only deal with this if boot.py doesn't exist and just use a generic boot.py
+        // For disabling, we can compare to the generic and if they are different refuse to touch it
+        const formElement = this.currentDialogElement.querySelector('#circuitpy_drive');
+        if (formElement) {
+            if (formElement.checked) {
+                this.tomlSettings['CIRCUITPY_DRIVE'] = "disabled";
```

Comment:
Should we add support to CP for this?

Reply:
Actually, that would make things much easier from the installer if we could. That was just some temporary stuff that's not currently being called.